lib/Widgets/SettingsPage/AbstractSettingsPage.vala

```diff
@@ -0,0 +1,35 @@
+/*
+* Copyright (c) 2017 elementary LLC. (https://elementary.io)
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published by
+* the Free Software Foundation, either version 2.1 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Library General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*/
+
+/**
+ * AbstractSettingsPage is a #Gtk.ScrolledWindow subclass with properties used
+ * by other Granite settings widgets.
+ */
+
+public abstract class Granite.SettingsPage : Gtk.ScrolledWindow {
+    /**
+     * An icon name associated with #this
+     */
+    public string? icon_name { get; construct set; }
```

Comment:
Now, I'm not sure about having only `icon_name` as a property that you can use to set an image for the page. That means that you cannot use anything like `Gdk.Pixbuf` or `GLib.Icon`. I would say, that there should be another property like a Pixbuf or one that's a `GLib.Icon`. Then you could have helper constructors like `from_icon_name` and `from_file` easily.